python/ray/data/dataset.py

```diff
@@ -868,12 +909,19 @@ def select_columns(
                 ray (e.g., num_gpus=1 to request GPUs for the map tasks).
         """  # noqa: E501
 
+        # Historically, we have also accepted lists with duplicate column names.
+        # This is not tolerated by the underlying pyarrow.Table.drop_columns method.
+        # Not that we can not use set() here because we want to preserve the order
+        cols_without_duplicates = list(dict.fromkeys(cols))
```

Comment:
I wanted to use that before but you can't use set() because it potentially changes the order so if you turn it to a list afterwards, the order of columns in the outputs will not be true to what user defined in inputs.

Reply:
I see, please capture this in a comment to make it clear what you're trying to achieve